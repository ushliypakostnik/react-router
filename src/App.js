import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Menu, { MenuItem } from './components/menu';
import classNames from "classnames";

import { StyleBase } from './scss/_stylebase.scss';

import './scss/layouts/_layout.scss';

let browserHistory = Router.browserHistory;

const PAGES = {
  page1: {
    path: "/page1",
    link: "Section 1",
    content: "CONTENT 1"
  },
  page2: {
    path: "/page2",
    link: "Section 2",
    content: "CONTENT 2"
  },
  page3: {
    path: "/page3",
    link: "Section 3",
    content: "CONTENT 3"
  }
};

const ARR = Object.values(PAGES).sort((a,b)=>a>b);

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      page: PAGES.page1
    };
  }

  switchPage(page) {
    this.setState({ page: page });
  }

  render() {
    return (
      <Router history={browserHistory}>
        <div className="page">
          <Menu>
            {ARR.map((item, index) => {
              return <MenuItem
                key={index}
                path={item.path}
                text={item.link}
                onClick={event => {
                  this.switchPage(item);
                }}
                className={this.state.page === item ? "page__menu--active" : ""}
              />
            })}
          </Menu>
          {ARR.map((item, index) => {
            return <Route
              key={index}
              path={item.path}
              component={Section}>
              <Section
                text={item.content}
                className={this.state.page === item ? "page__section--active" : ""}
              />
            </Route>
          })}
        </div>
      </Router>
    );
  }
}

class Section extends Component {
  render() {
    return (
      <section className={classNames("page__section", this.props.className)} >
        {this.props.text}
      </section>
    );
  }
}

export default App;
