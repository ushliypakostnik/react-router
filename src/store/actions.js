const fetchUrl = process.env.REACT_APP_API_URL;

export const pageToActive = (page) => ({
  type: "PAGE_TO_ACTIVE",
  activePage: page
});

export const setMinHeight = (minHeight) => ({
  type: "SET_MIN_HEIGHT",
  minHeight: minHeight
});

export const toogleTheme = (theme) => ({
  type: "TOGGLE_THEME",
  theme: theme
});

export const requestAlbums = () => ({
  type: "REQUEST_ALBUMS"
});

export const receiveAlbums = (albums) => ({
  type: "RECEIVE_ALBUMS",
  albums
});

export const requestAlbumFailed = (error) => ({
  type: "REQUEST_ALBUMS_FAILED",
  error
});

export const fetchAlbums = () => {
  return dispatch => {
    dispatch(requestAlbums());
    return fetch(fetchUrl + "/albums")
      .then(res => res.json())
      .then(
        (result) => {
          /* eslint-disable array-callback-return */
          let resultArr = [];
          result.map((item, index) => {
            if (index === 0) {
              resultArr.push({text: item.name, path: "/"});
            } else {
              resultArr.push({text: item.name, path: "/" + item.id});
            }
          });
          dispatch(receiveAlbums(resultArr));
        },
        (error) => {
          dispatch(requestAlbumFailed(error));
        }
      );
  }
}

export const requestData = () => ({
  type: "REQUEST_DATA"
});

export const receiveData = (data) => ({
  type: "RECEIVE_DATA",
  data
});

export const requestDataFailed = (error) => ({
  type: "REQUEST_DATA_FAILED",
  error
});

export const fetchData = (album) => {
  return dispatch => {
    dispatch(requestData());
    return fetch(fetchUrl + "/albums/" + album)
      .then(res => res.json())
      .then(
        (result) => {
          dispatch(receiveData(result));
        },
        (error) => {
          dispatch(requestDataFailed(error));
        }
      );
  }
}
