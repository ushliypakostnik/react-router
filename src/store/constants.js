export const FETCH_URL = process.env.REACT_APP_API_URL || 'http://127.0.0.1:8082';

export const THEME = {
  DARK: 'dark',
  LIGHT: 'light'
};

export const COOKIES = {
  THEME: 'theme'
};

export const INITIAL_STATE = {
  reducer: {
    activePage: '/',
    theme: THEME.DARK,
    isFetching: false,
    albums: [],
    data: [],
    resize: {
      minHeight: 'auto',
      deviceType: 'large'
    },
    error: null
  },
  routing: {}
};

export const SCREENS = {
  MOBILE: 'mobile',
  MOBILE_2X: 'mobile-2x',
  DESKTOP: 'desktop',
  DESKTOP_2X: 'desktop-2x'
};
