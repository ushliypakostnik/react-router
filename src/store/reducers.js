const reducer = (state = { }, action) => {
  switch (action.type) {
    case "REQUEST_ALBUMS":
      return Object.assign({}, state, {
        isFetching: true
      });
    case "RECEIVE_ALBUMS":
      return Object.assign({}, state, {
        isFetching: false,
        albums: action.albums
      });
    case "REQUEST_ALBUMS_FAILED":
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });
    case "REQUEST_DATA":
      return Object.assign({}, state, {
        isFetching: true
      });
    case "RECEIVE_DATA":
      return Object.assign({}, state, {
        isFetching: false,
        data: action.data
      });
    case "REQUEST_DATA_FAILED":
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });
    case "PAGE_TO_ACTIVE":
      return Object.assign({}, state, {
        activePage: action.activePage
      });
    case "TOGGLE_THEME":
      return Object.assign({}, state, {
        theme: action.theme
      });
    default:
      return state;
  }
}

export default reducer;
