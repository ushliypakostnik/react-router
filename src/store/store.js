import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { browserHistory } from "react-router";
//import { createBrowserHistory } from "history";
import { syncHistoryWithStore, routerReducer } from 'react-router-redux';

import { INITIAL_STATE } from './constants';
import { pageToActive } from './actions';
import reducer from './reducers';

const middlewares = [];
middlewares.push(thunkMiddleware)

if (process.env.NODE_ENV !== 'production') {
  const loggerMiddleware = createLogger();

  middlewares.push(loggerMiddleware);
}

function configureStore(state) {
  return createStore(
    combineReducers({
      reducer,
      routing: routerReducer
    }),
    state,
    applyMiddleware(...middlewares)
  );
}

const store = configureStore(INITIAL_STATE);

//const history = createBrowserHistory();
export const history = syncHistoryWithStore(browserHistory, store);
//history.listen(location => console.info('-> location:', location));

let lastUrl = "";
if (store.getState().routing.locationBeforeTransitions != null) {
  lastUrl = store.getState().routing.locationBeforeTransitions.pathname;
} else {
  lastUrl = "/";
}

//store.subscribe(() => {
//  console.log("Store: ", store.getState());
//  console.log("last Url: ", lastUrl);
//});

store.dispatch(pageToActive(lastUrl));

export default store;
