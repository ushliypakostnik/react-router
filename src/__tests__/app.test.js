import React from 'react';
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { FETCH_URL, INITIAL_STATE } from '../store/constants';
import { fetchAlbums } from '../store/actions';

import App from '../containers/app';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('app render', () => {

  let store,
      wrapper;

  beforeEach(() => {
    store = mockStore(INITIAL_STATE);
    wrapper = mount(<Provider store={store}>
                      <MemoryRouter>
                        <App />
                      </MemoryRouter>
                    </Provider>);
  });

  it('app render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('app has wrapper with class', () => {
    expect(wrapper.find('.app')).toHaveLength(1);
  });

  it('app has resize', () => {
    expect(wrapper.find('Resize')).toHaveLength(1);
  });

  it('app has header', () => {
    expect(wrapper.find('Header')).toHaveLength(1);
  });

  it('app has switch', () => {
    expect(wrapper.find('Switch')).toHaveLength(1);
  });

  afterEach(() => wrapper.unmount());
});
