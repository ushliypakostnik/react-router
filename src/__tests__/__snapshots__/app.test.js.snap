// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app render app render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "k5q3wz",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "k5q3wz",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(App)>
        <App
          albums={Array []}
          fetchAlbums={[Function]}
        >
          <div
            className="app"
          >
            <Connect(Resize)>
              <Resize
                resize={[Function]}
              >
                <div
                  className="app__resize"
                >
                  <ResizeDetector
                    handleHeight={true}
                    handleWidth={false}
                    nodeType="div"
                    onResize={[Function]}
                    querySelector={null}
                    refreshRate={1000}
                    skipOnMount={false}
                    targetDomEl={null}
                  >
                    <ChildWrapper>
                      <div />
                    </ChildWrapper>
                  </ResizeDetector>
                </div>
              </Resize>
            </Connect(Resize)>
            <Connect(Header)
              items={Array []}
            >
              <Header
                deviceType="large"
                items={Array []}
                pageIsActive="/"
                pageToActive={[Function]}
              >
                <header
                  className="app__header header"
                  role="banner"
                >
                  <div
                    className="container-fluid"
                  >
                    <nav
                      role="navigation"
                    >
                      <Menu>
                        <ul
                          className="app__menu"
                        />
                      </Menu>
                    </nav>
                    <a
                      aria-label="Open Menu"
                      className="header__navbar"
                      href="#"
                      onClick={[Function]}
                    >
                      <Icon
                        aria-hidden="true"
                        type="menu"
                      >
                        <LocaleReceiver
                          componentName="Icon"
                        >
                          <i
                            aria-hidden="true"
                            aria-label="icon: menu"
                            className="anticon anticon-menu"
                          >
                            <IconReact
                              className=""
                              type="menu-o"
                            >
                              <svg
                                aria-hidden="true"
                                className=""
                                data-icon="menu"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                key="svg-menu"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                                  key="svg-menu-svg-0"
                                />
                              </svg>
                            </IconReact>
                          </i>
                        </LocaleReceiver>
                      </Icon>
                    </a>
                    <withConfigConsumer(Drawer)
                      className="app__panel"
                      closable={true}
                      onClose={[Function]}
                      placement="right"
                      title={null}
                      visible={false}
                      width="40%"
                    >
                      <Drawer
                        className="app__panel"
                        closable={true}
                        getPrefixCls={[Function]}
                        height={256}
                        level={null}
                        mask={true}
                        maskClosable={true}
                        onClose={[Function]}
                        placement="right"
                        prefixCls="ant-drawer"
                        renderEmpty={[Function]}
                        title={null}
                        visible={false}
                        width="40%"
                      >
                        <Drawer
                          afterVisibleChange={[Function]}
                          className="app__panel"
                          closable={true}
                          duration=".3s"
                          ease="cubic-bezier(0.78, 0.14, 0.15, 0.86)"
                          firstEnter={false}
                          getContainer="body"
                          getPrefixCls={[Function]}
                          handler={false}
                          keyboard={true}
                          level={null}
                          mask={true}
                          maskClosable={true}
                          maskStyle={Object {}}
                          onChange={[Function]}
                          onClose={[Function]}
                          onMaskClick={[Function]}
                          open={false}
                          placement="right"
                          prefixCls="ant-drawer"
                          renderEmpty={[Function]}
                          showMask={true}
                          style={
                            Object {
                              "transform": undefined,
                              "zIndex": undefined,
                            }
                          }
                          title={null}
                          visible={false}
                          width="40%"
                          wrapperClassName=""
                        />
                      </Drawer>
                    </withConfigConsumer(Drawer)>
                    <address
                      className="header__right"
                      role="contentinfo"
                    >
                      <Connect(Component)>
                        <ForwardRef
                          theme="dark"
                          toggleTheme={[Function]}
                        >
                          <withCookies(ThemeSwitch)
                            cookies={
                              Cookies {
                                "HAS_DOCUMENT_COOKIE": true,
                                "changeListeners": Array [
                                  [Function],
                                ],
                                "cookies": Object {},
                              }
                            }
                            forwardedRef={null}
                            theme="dark"
                            toggleTheme={[Function]}
                          >
                            <ThemeSwitch
                              allCookies={Object {}}
                              cookies={
                                Cookies {
                                  "HAS_DOCUMENT_COOKIE": true,
                                  "changeListeners": Array [
                                    [Function],
                                  ],
                                  "cookies": Object {},
                                }
                              }
                              theme="dark"
                              toggleTheme={[Function]}
                            >
                              <a
                                aria-label="Theme Switch"
                                className="app__theme-switch theme-switch"
                                href="#"
                                onClick={[Function]}
                              >
                                <span
                                  aria-hidden="true"
                                  className="theme-switch__ico"
                                />
                                <span
                                  className="theme-switch__label"
                                >
                                  light
                                   theme
                                </span>
                              </a>
                            </ThemeSwitch>
                          </withCookies(ThemeSwitch)>
                        </ForwardRef>
                      </Connect(Component)>
                      <span
                        className="header__logo header__logo--xs"
                      >
                        Ivan Samovarov
                      </span>
                      <a
                        aria-label="Ivan Samovarov livejournal"
                        className="header__social header__social--lj"
                        href="https://sambus.livejournal.com"
                        target="_blank"
                      >
                        <LjIcon
                          aria-hidden="true"
                        >
                          <svg
                            aria-hidden="true"
                            focusable="false"
                            role="img"
                            viewBox="0 0 100 100"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M10.941,34.24C23.797,47.097,36.52,59.819,49.25,72.551 c4.946-10.602,12.749-18.402,23.357-23.35c-12.734-12.734-25.459-25.459-38.232-38.23C47.06,4.332,70.25,3.879,86.787,20.768 c16.847,17.204,17.686,44.49,1.933,62.94c-15.611,18.285-42.869,21.59-62.516,7.463C6.053,76.684,2.741,51.459,10.941,34.24z M75.545,60.745c-6.476,3.439-11.302,8.271-14.757,14.752c6.151,1.208,12.16,2.387,18.356,3.604 C77.932,72.911,76.752,66.899,75.545,60.745z M23.881,0.29c-0.249-0.243-0.958-0.284-1.31-0.117 C12.489,4.977,5.043,12.438,0.183,22.495c-0.318,0.659-0.222,1.054,0.285,1.55c2.419,2.371,4.812,4.768,7.189,7.18 c0.983,0.996,1.907,2.051,3.015,3.247c5.117-10.923,12.91-18.724,23.684-23.768C30.757,7.107,27.344,3.673,23.881,0.29z"
                            />
                          </svg>
                        </LjIcon>
                      </a>
                      <a
                        aria-label="Ivan Samovarov instagram"
                        className="header__social"
                        href="https://www.instagram.com/samovarov_ivan"
                        target="_blank"
                      >
                        <FontAwesomeIcon
                          aria-hidden="true"
                          border={false}
                          className=""
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                448,
                                512,
                                Array [],
                                "f16d",
                                "M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z",
                              ],
                              "iconName": "instagram",
                              "prefix": "fab",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-instagram fa-w-14 "
                            data-icon="instagram"
                            data-prefix="fab"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 448 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                      </a>
                      <a
                        aria-label="Ivan Samovarov facebook"
                        className="header__social"
                        href="https://www.facebook.com/samovaru"
                        target="_blank"
                      >
                        <FontAwesomeIcon
                          aria-hidden="true"
                          border={false}
                          className=""
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                320,
                                512,
                                Array [],
                                "f39e",
                                "M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z",
                              ],
                              "iconName": "facebook-f",
                              "prefix": "fab",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-facebook-f fa-w-10 "
                            data-icon="facebook-f"
                            data-prefix="fab"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 320 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                      </a>
                      <a
                        aria-label="Ivan Samovarov vkontakte"
                        className="header__social header__social--vk"
                        href="https://vk.com/samovaru"
                        target="_blank"
                      >
                        <FontAwesomeIcon
                          aria-hidden="true"
                          border={false}
                          className=""
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                576,
                                512,
                                Array [],
                                "f189",
                                "M545 117.7c3.7-12.5 0-21.7-17.8-21.7h-58.9c-15 0-21.9 7.9-25.6 16.7 0 0-30 73.1-72.4 120.5-13.7 13.7-20 18.1-27.5 18.1-3.7 0-9.4-4.4-9.4-16.9V117.7c0-15-4.2-21.7-16.6-21.7h-92.6c-9.4 0-15 7-15 13.5 0 14.2 21.2 17.5 23.4 57.5v86.8c0 19-3.4 22.5-10.9 22.5-20 0-68.6-73.4-97.4-157.4-5.8-16.3-11.5-22.9-26.6-22.9H38.8c-16.8 0-20.2 7.9-20.2 16.7 0 15.6 20 93.1 93.1 195.5C160.4 378.1 229 416 291.4 416c37.5 0 42.1-8.4 42.1-22.9 0-66.8-3.4-73.1 15.4-73.1 8.7 0 23.7 4.4 58.7 38.1 40 40 46.6 57.9 69 57.9h58.9c16.8 0 25.3-8.4 20.4-25-11.2-34.9-86.9-106.7-90.3-111.5-8.7-11.2-6.2-16.2 0-26.2.1-.1 72-101.3 79.4-135.6z",
                              ],
                              "iconName": "vk",
                              "prefix": "fab",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-vk fa-w-18 "
                            data-icon="vk"
                            data-prefix="fab"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 576 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M545 117.7c3.7-12.5 0-21.7-17.8-21.7h-58.9c-15 0-21.9 7.9-25.6 16.7 0 0-30 73.1-72.4 120.5-13.7 13.7-20 18.1-27.5 18.1-3.7 0-9.4-4.4-9.4-16.9V117.7c0-15-4.2-21.7-16.6-21.7h-92.6c-9.4 0-15 7-15 13.5 0 14.2 21.2 17.5 23.4 57.5v86.8c0 19-3.4 22.5-10.9 22.5-20 0-68.6-73.4-97.4-157.4-5.8-16.3-11.5-22.9-26.6-22.9H38.8c-16.8 0-20.2 7.9-20.2 16.7 0 15.6 20 93.1 93.1 195.5C160.4 378.1 229 416 291.4 416c37.5 0 42.1-8.4 42.1-22.9 0-66.8-3.4-73.1 15.4-73.1 8.7 0 23.7 4.4 58.7 38.1 40 40 46.6 57.9 69 57.9h58.9c16.8 0 25.3-8.4 20.4-25-11.2-34.9-86.9-106.7-90.3-111.5-8.7-11.2-6.2-16.2 0-26.2.1-.1 72-101.3 79.4-135.6z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                      </a>
                      <span
                        className="header__logo"
                      >
                        Ivan Samovarov
                      </span>
                    </address>
                  </div>
                </header>
              </Header>
            </Connect(Header)>
            <Switch />
          </div>
        </App>
      </Connect(App)>
    </Router>
  </MemoryRouter>
</Provider>
`;
